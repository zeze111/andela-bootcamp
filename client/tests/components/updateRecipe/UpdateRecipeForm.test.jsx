import React from 'react';

import connectedUpdateRecipeForm, {UpdateRecipeForm} from '../../../components/updateRecipe/UpdateRecipeForm';
import store from '../../../store';

const props = {
  updateRecipe: jest.fn(() => Promise.resolve()),
  recipe: {
    id: 3,
    name: 'something'
  }
}

const setup = () => {
  const wrapper = shallow(<UpdateRecipeForm {...props} store={store} />);
  return { wrapper }
}

describe('component will receive props', () => {
  it('should call get recipes function', () => {
    const spy = sinon.spy(UpdateRecipeForm.prototype, 'componentWillReceiveProps');
  });
});

describe('onChange', () => {
  it('should change event', () => {
    const event = {
      target: { name: 'name', value: 'Rice and stew' },
    };
    const { wrapper } = setup();
    const recipeName = wrapper.find('#uname');
    recipeName.simulate('change', event);
    expect(wrapper.instance().state.name).toBe('Rice and stew');
  });
});

describe('onSelectChange', () => {
  it('should change event for type select field', () => {
    const event = {
      target: { value: 'Main' },
    };
    const { wrapper } = setup();
    const type = wrapper.find('#utype');
    type.simulate('change', event);
    expect(wrapper.instance().state.type).toBe('Main');
  });
});

describe('onSubmit', () => {
  it('be called when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.setState({
      name: 'Rice and stew',
      ingredients: 'rice, tomatoes, pepper, onions',
      instructions: 'Boil the rice',
      imageFile: { name: 'pic.jpg' },
      preparationTime: '12 minutes',
      type: 'Main',
      errors: {},
      isLoading: false
    });

    wrapper.instance().onSubmit(event);
    expect(props.updateRecipe.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return validation error when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };

    wrapper.setState({
      name: 'R'
    });
    wrapper.instance().onSubmit(event);
    expect(props.updateRecipe.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return an error when form is submitted', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    props.updateRecipe = jest.fn(() => Promise.reject());
    const wrapper = shallow(<UpdateRecipeForm {...props} store={store} />);

    wrapper.setState({
      name: 'Rice and stew',
      ingredients: 'rice, tomatoes, pepper, onions',
      instructions: 'Boil the rice',
      imageFile: { name: 'pic.jpg' },
      preparationTime: '12 minutes',
      type: 'Main',
      errors: {},
      isLoading: false
    });
    wrapper.instance().onSubmit(event);
    expect(props.updateRecipe.mock.calls.length).toEqual(1);
  });
});

describe('Connect Update Recipe component', () => {
  it('should render component successfully', () => {
    const store = mockStore({
        recipeReducer: {
          recipes: []
        }
    })

    const wrapper = shallow(<UpdateRecipeForm {...props} store={store} />);
    expect(wrapper.length).toBe(1);
  });
});

describe('Test for Update Recipe Form', () => {
  it('should render component', () => {

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
