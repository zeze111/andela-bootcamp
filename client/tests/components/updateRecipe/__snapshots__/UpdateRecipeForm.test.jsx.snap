// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Update Recipe Form should render component 1`] = `
Object {
  "wrapper": <div
    className="row"
  >
    <div
      className="card-content col s10 m8 l6 push-s1 push-m2 push-l3"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="row add-padding"
        >
          <div
            className="file-field input-field col s12 right-align get-pic-btn"
          >
            <div
              className="image-placeholder"
            >
              <img
                alt=""
                className="recipe-image"
                src="/images/noimg.png"
              />
            </div>
            <div
              className="btn-floating btn-small waves-effect waves-light blue"
            >
              <i
                className="material-icons"
              >
                photo
              </i>
              <input
                onChange={[Function]}
                type="file"
              />
            </div>
          </div>
        </div>
        <div
          className="half-top"
        >
          <span
            className="red-text error-text"
          />
          <TextFieldGroup
            active="active"
            error={Array []}
            id="uname"
            label="Recipe Name"
            name="name"
            onChange={[Function]}
            placeholder=""
            req=""
            type="text"
            value=""
          />
          <TextFieldGroup
            active="active"
            error={Array []}
            id="utime"
            label="Preparation Time"
            name="preparationTime"
            onChange={[Function]}
            placeholder="e.g. 1 hour / 2hrs 30mins"
            req=""
            type="text"
            value=""
          />
          <TextFieldGroup
            active="active"
            error={Array []}
            id="udesc"
            label="Description"
            name="description"
            onChange={[Function]}
            placeholder=""
            req=""
            type="text"
            value=""
          />
          <Row>
            <Input
              defaultValue="0"
              id="utype"
              label="Recipe Type"
              name="type"
              onChange={[Function]}
              s={12}
              type="select"
            >
              <option
                disabled={true}
                value="0"
              >
                 Choose your option 
              </option>
              <option
                value="Appetizer"
              >
                Appetizer
              </option>
              <option
                value="Main"
              >
                Main
              </option>
              <option
                value="Dessert"
              >
                Dessert
              </option>
              <option
                value="Drinks"
              >
                Drinks
              </option>
            </Input>
            <span
              className="help-block red-text error-text2"
            />
          </Row>
          <TextFieldGroup2
            error={Array []}
            id="uingred"
            label="Ingredients"
            name="ingredients"
            onChange={[Function]}
            placeholder=""
            req=""
            value=""
          />
          <TextFieldGroup2
            error={Array []}
            id="uinstruct"
            label="Instructions"
            name="instructions"
            onChange={[Function]}
            placeholder="e.g. Preheat oven at 120 celcius. boil egg."
            req=""
            value=""
          />
          <div
            className="right-align"
          >
            <input
              className="btn grey"
              disabled={false}
              type="submit"
              value="Submit"
            />
          </div>
           
          <br />
        </div>
      </form>
    </div>
  </div>,
}
`;
